File: Go/004_Program/main.go
package main

import (
    "fmt"
    "os"
)

func main() {
    fmt.Println("=== File Writer ===")

    var filename string
    fmt.Print("Enter filename: ")
    fmt.Scanln(&filename)

    fmt.Println("Enter text (type 'END' on a new line to finish):")

    file, err := os.Create(filename)
    if err != nil {
        fmt.Printf("Error: %v\n", err)
        return
    }
    defer file.Close()

    var line string
    lineCount := 0

    for {
        fmt.Scanln(&line)
        if line == "END" {
            break
        }
        _, err := file.WriteString(line + "\n")
        if err != nil {
            fmt.Printf("Error writing: %v\n", err)
            return
        }
        lineCount++
    }

    fmt.Printf("\nWrote %d lines to %s\n", lineCount, filename)
}

File: Rust/004_Program/main.rs
use std::fs::File;
use std::io::{self, Write};

fn main() {
    println!("=== File Writer ===");

    print!("Enter filename: ");
    io::stdout().flush().unwrap();

    let mut filename = String::new();
    io::stdin().read_line(&mut filename).unwrap();
    let filename = filename.trim();

    match File::create(filename) {
        Ok(mut file) => {
            println!("Enter text (type 'END' to finish):");

            loop {
                let mut line = String::new();
                io::stdin().read_line(&mut line).unwrap();

                if line.trim() == "END" {
                    break;
                }

                if let Err(e) = file.write_all(line.as_bytes()) {
                    eprintln!("Error writing: {}", e);
                    return;
                }
            }

            println!("File written successfully");
        }
        Err(e) => eprintln!("Error creating file: {}", e),
    }
}
